File: internal/content/testdata/bench/go/nc-kth-smallest-bst.go
func kth_smallest(root *TreeNode, k int) int {
	stack := []*TreeNode{}
	cur := root
	for len(stack) > 0 || cur != nil {
		for cur != nil {
			stack = append(stack, cur)
			cur = cur.Left
		}
		cur = stack[len(stack)-1]
		stack = stack[:len(stack)-1]
		k--
		if k == 0 {
			return cur.Val
		}
		cur = cur.Right
	}
	return -1
}

File: internal/content/testdata/bench/rust/nc-kth-smallest-bst.rs
use std::rc::Rc;
use std::cell::RefCell;

fn kth_smallest(root: Option<Rc<RefCell<TreeNode>>>, k: i64) -> i64 {
    let mut stack: Vec<Rc<RefCell<TreeNode>>> = Vec::new();
    let mut cur = root;
    let mut k = k;
    while !stack.is_empty() || cur.is_some() {
        while let Some(n) = cur {
            let left = n.borrow().left.clone();
            stack.push(n);
            cur = left;
        }
        let node = stack.pop().unwrap();
        k -= 1;
        if k == 0 {
            return node.borrow().val as i64;
        }
        cur = node.borrow().right.clone();
    }
    -1
}
